test/test_common/utility.h

```diff
@@ -757,13 +757,6 @@ template <class Interface, class Impl> class TestHeaderMapImplBase : public Inte
     }
     header_map_->verifyByteSizeInternalForTest();
   }
-  TestHeaderMapImplBase(
```

Comment:
This was introduced by https://github.com/envoyproxy/envoy/pull/11546 cc @mattklein123 in case you have more context/if we should not remove this, the PR description has a godbolt.org link that shows the same issue we see on Windows